mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java

```diff
@@ -87,6 +92,29 @@ private void setupUi() {
         vStateView.setVisibility(View.GONE);
     }
 
+    @Override
+    public void scrollDownBottom() {
+        rvHistory.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
+                if (!recyclerView.canScrollVertically(1)) {
+                    Toast.makeText(getContext(), R.string.end_of_transactions,
+                            Toast.LENGTH_SHORT).show();
+                } else {
+                    scrollDownButton.setVisibility(View.VISIBLE);
+                }
+            }
+        });
+        scrollDownButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                int length = mHistoryAdapter.getItemCount();
+                rvHistory.scrollToPosition(length - 1);
```

Comment:
Just a suggestion here :) 
Although I have not seen this in action in the actual app, but generally scrollToPosition leads to abrupt change in the UI. 
I would suggest trying smoothScrollToPosition which leads to decent animation while scrolling to the bottom.

Reply:
Okay!